MonoGame.Framework/Graphics/Vertices/VertexBufferBindings.cs

```diff
@@ -88,22 +88,33 @@ public bool Set(VertexBuffer vertexBuffer, int vertexOffset)
             return true;
         }
 
+
+        /// <inheritdoc cref="Set(VertexBufferBinding[], int)"/>
+        public bool Set(params VertexBufferBinding[] vertexBufferBindings)
+        {
+            return Set(vertexBufferBindings, vertexBufferBindings.Length);
+        }
+
+        ///
         /// <summary>
         /// Binds the the specified vertex buffers to the input slots.
         /// </summary>
         /// <param name="vertexBufferBindings">The vertex buffer bindings.</param>
+        /// <param name="bindingCount">Number of bindings from the array to set, starting from index 0.</param>
         /// <returns>
         /// <see langword="true"/> if the input layout was changed; otherwise,
         /// <see langword="false"/>.
         /// </returns>
-        public bool Set(params VertexBufferBinding[] vertexBufferBindings)
+        public bool Set(VertexBufferBinding[] vertexBufferBindings, int bindingCount)
```

Comment:
May be add doc on the negative effects of passing a binding (and its effect on per-frame heap alloc).
IMO, I feel like the API user can do all this without introducing `_vertexBufferBindings[]` array at the `GraphicsDevice` level - you can maintain it yourself, can't you? But I realize you already made some `params` changes so will let others chime in here.

It's a good idea though, but could mostly live inside a doc comment as opposed to prescribing/adding complexity..